implementations/rust/ockam/ockam_command/src/relay/create.rs

```diff
@@ -37,8 +37,8 @@ pub struct CreateCommand {
     relay_name: String,
 
     /// Node for which to create the relay
-    #[arg(long, id = "NODE", display_order = 900, default_value_t = default_node_name(), value_parser = node_name_parser)]
-    to: String,
+    #[arg(long, id = "NODE", display_order = 900)]
+    to: Option<String>,
```

Comment:
I have a concern here.

And below with line #44.

From what I can tell, this will undo the changes that enable commands to create a required resource if needed.

Perhaps we can update the get_node_name/cli_state to allow for creating a resource on demand if one is required.

Reply:
To add to what Adrian said above. I do agree, I think it's great that the parsers and default values in this case are going away.

There's a few wins to be gained from moving it away from there; however it's still a functionality we should re-introduce.

Reply:
@CAOakleyII the behaviour should be exactly the same as before if you have a look at the implementation for `get_node_name`. It uses `default_node_name` if there is no value provided and `node_name_parser` if a value is given. The only difference is that those 2 functions now accept a `CliState` as their argument instead of recreating it with `try_default`

Reply:
BTW I have nothing at all against default values and parsers, on the contrary! I think it's better if we can parse user-input strings directly as proper types (`IdentityIdentifier`, `PublicKey` etc...) but not if that means triggering side-effects containing business logic or touching the file system.

Reply:
<img width="454" alt="image" src="https://github.com/build-trust/ockam/assets/1088854/a03c6fb2-302f-4bc6-9f0f-dd3269f3c575">

Reply:
The issue is before, it would call the "default_node_name" first, and if there's no "default node" it would return the name "default"


node_name_parser, then would be called after, and again if there's no "default node" it would create the node next.

Reply:
Thanks @CAOakleyII I get it now. Please see my latest commits where:

 - I call `initialize_node_if_default` and `initialize_identity_if_default` before running an async command (because those 2 functions might call another command and we cannot nest commands)
 
 - I added unit tests to specify the expected behaviour for that piece of code